connection.cc

```diff
@@ -40,4 +40,6 @@ void Connection::send(std::size_t length) {
           receive();
```

Comment:
would that mean we would close the socket as soon as we get a response? 
Connection::send() will hang until the callback receives something, right?